source/server/http/BUILD

```diff
@@ -10,8 +10,16 @@ envoy_package()
 
 envoy_cc_library(
     name = "admin_lib",
-    srcs = ["admin.cc"],
-    hdrs = ["admin.h"],
+    srcs = [
+        "admin.cc",
```

Comment:
split out admin_filter.cc into its own library?

I think you don't have a dependence from admin.h/admin.cc/admin_test.cc to admin_filter*. In general in Envoy we've been keeping libraries relatively fine grained.

Reply:
There is a dependency between `AdminImpl` and `AdminFilter` actually https://github.com/envoyproxy/envoy/blob/9186ebf374b1bd7d7c65ed1cec748377f88ec0a7/source/server/http/admin.cc#L1640-L1643
Having said that, I can still split out the admin_filter.cc into its own lib, if that better aligns with the general approach adopted for envoy. Wdyt?

Reply:
I see...as long as the two files are codependent, I'm pretty sure you'll need a single library for them; Bazel won't let you have mutually dependent libraries.

We *could* break the dependence from AdminFilter to Admin via a lambda, e.g in that block:

 AdminFilter filter([this](absl::string_view& path,  Http::ResponseHeaderMap& header, Buffer::OwnedImpl& response, AdminFilter& filter) -> Http::Code { 
  return runCallback(path, *header_map, response, *this);
});

or similar.

This would also make AdminFilter individually testable without instantiating an Admin.

WDYT?

Reply:
> Bazel won't let you have mutually dependent libraries.

Oh, I see. I think that breaking the circular dependency is a good idea: I will go ahead with the lambda approach that you suggested.